pkg/webhook/webhook_test.go

```diff
@@ -58,3 +60,62 @@ func TestBuildFleetGuardRailValidatingWebhooks(t *testing.T) {
 		})
 	}
 }
+
+func TestNewWebhookConfig(t *testing.T) {
+	tests := []struct {
+		name                          string
+		mgr                           manager.Manager
+		webhookServiceName            string
+		port                          int32
+		clientConnectionType          *options.WebhookClientConnectionType
+		certDir                       string
+		enableGuardRail               bool
+		denyModifyMemberClusterLabels bool
+		want                          *Config
+		wantErr                       bool
+	}{
+		{
+			name:                          "valid input",
+			mgr:                           nil,
+			webhookServiceName:            "test-webhook",
+			port:                          8080,
+			clientConnectionType:          nil,
+			certDir:                       "/tmp/cert",
+			enableGuardRail:               true,
+			denyModifyMemberClusterLabels: true,
+			want: &Config{
+				serviceNamespace:              "test-namespace",
+				serviceName:                   "test-webhook",
+				servicePort:                   8080,
+				clientConnectionType:          nil,
+				enableGuardRail:               true,
+				denyModifyMemberClusterLabels: true,
+			},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Setenv("POD_NAMESPACE", "test-namespace")
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := NewWebhookConfig(tt.mgr, tt.webhookServiceName, tt.port, tt.clientConnectionType, tt.certDir, tt.enableGuardRail, tt.denyModifyMemberClusterLabels)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("NewWebhookConfig() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if got == nil || tt.want == nil {
+				if got != tt.want {
+					t.Errorf("NewWebhookConfig() = %v, want %v", got, tt.want)
+				}
+				return
+			}
+			if got.serviceNamespace != tt.want.serviceNamespace ||
```

Comment:
Oh you can do:
`if diff := cmp.Diff(got, tt.want); diff != "" {fmt.Errorf("...-got, +want: diff)}`
https://github.com/kubefleet-dev/kubefleet/blob/afb85c4d453e52b8893d0b14b16b31ef25424a92/test/scheduler/actuals_test.go#L139C3-L141C4
No need to compare them one by one anymore.

Reply:
Oh you can ignore that field by adding cmpOption: https://github.com/kubefleet-dev/kubefleet/blob/afb85c4d453e52b8893d0b14b16b31ef25424a92/test/e2e/v1alpha1/e2e_test.go#L94C2-L99C3